### DISCUSSION TITLE
State of Terraform and OpenTF

### DISCUSSION THREAD

**Discussion by codeagencybe** [mysticaltech/terraform-hcloud-kube-hetzner#976] [Ideas]
@mysticaltech 
Just opening the discussion about how Kube-Hetzner moves forward in the current "hot" discussion about Terraform changing it's license to BSL. 

Simple put/take away: if you build any service on top of Kubernetes, you can no longer use Terraform free of charge. 

Some people may use this project for personal projects, selfhosting stuff etc.. but some people build a company business on top of this by offering hosting to clients. 

AFAIK and understand, it seems like the goal of OpenTF is to be a full drop-in replacement with the last version of Terraform that was still under open license. 
Any providers like Hetzner that work under Terraform should also just work OOTB with OpenTF but I suppose it means (a lot?) changes in the project if it wants to remain OSS? 

Any thoughs, input on this topic?  

https://opentf.org/

The fork is now also officially available and public at open fork: https://github.com/opentffoundation/opentf

It's still early days at the moment, but it's important to make a decision early on this. 
A lot of companies who build services on top of the open source version of Terraform, already feel the costs slapping on them hard now. 
I don't know if Terraform (as in the company Hashicorp) has ways to find uses of there of tooling into the wild and hunt down companies with bills, but I would be cautious with the situation now.

**Comment by mysticaltech** [comment]
@codeagencybe Yep, absolutely! Thanks for sharing. We've also been monitoring the situation. But it's good to have an public discussion here.

For now just observing, but definitely we will stick with opentf.
